packages/gas-fee-controller/src/GasFeeController.ts

```diff
@@ -319,48 +317,30 @@ export class GasFeeController extends BaseControllerV2<
     this.pollTokens = new Set();
     this.getCurrentNetworkEIP1559Compatibility =
       getCurrentNetworkEIP1559Compatibility;
-
     this.getCurrentNetworkLegacyGasAPICompatibility =
       getCurrentNetworkLegacyGasAPICompatibility;
-
     this.getCurrentAccountEIP1559Compatibility =
       getCurrentAccountEIP1559Compatibility;
+    this.#getProvider = getProvider;
     this.EIP1559APIEndpoint = EIP1559APIEndpoint;
     this.legacyAPIEndpoint = legacyAPIEndpoint;
     this.clientId = clientId;
+
+    this.ethQuery = new EthQuery(this.#getProvider());
```

Comment:
We don't need to call `getProvider` only if `onNetworkStateChange` and `getChainId` are given; this is a required option, so we can always get the provider this way. So I chose to use `getProvider` in the state change callback below to refresh the EthQuery instance. However, if we want to continue using the `getEthQuery` action we can — I just didn't know if we planned on getting rid of that or not.